src/sundials_matrix_wrapper.cpp

```diff
@@ -100,6 +100,31 @@ operator=(SUNMatrixWrapper &&other) {
     return *this;
 }
 
+void SUNMatrixWrapper::reallocate(int NNZ) {
```

Comment:
having both `realloc` and `reallocate` with different purposes seems to be asking for trouble... not sure about better naming though...

Reply:
thats the sundials naming scheme ... since realloc has no int argument, I would hope confusion shouldn't be too big.